app/src/main/java/me/iacn/biliroaming/hook/JsonHook.kt

```diff
@@ -65,15 +65,25 @@ class JsonHook(classLoader: ClassLoader) : BaseHook(classLoader) {
                             }
                         }
                     }
-                    if (sPrefs.getBoolean("purify_home_tab", false)) {
+
+                    if (sPrefs.getStringSet("purify_home_tab", emptySet())?.isNotEmpty() == true) {
                         val tab = data?.getObjectFieldAs<MutableList<Any>>("tab")
+                        val purifytabset = sPrefs.getStringSet("purify_home_tab", emptySet())
                         tab?.removeAll {
                             it.getObjectFieldAs<String>("uri").run {
```

Comment:
`when(it.getObjectFieldAs<String>("uri")) {}`

Reply:
in this way "other tabs" section will not work, are you sure want me to do that

Reply:
Then switch to the strategy that stores regex in the entries.

Reply:
seems like i had misunderstood what you said before, but i understand now.
if i use `when(it.getObjectFieldAs<String>("uri")) {}` then i can't get value by `this` which `when` is using, cause of that `startwith` and regex will not work